test/tests/link.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const { ariaTest } = require('..');
+
+let pageExamples = [
+  {
+    exampleId: 'ex1',
+    linkSelector: '#ex1 span'
+  },
+  {
+    exampleId: 'ex2',
+    linkSelector: '#ex2 img',
+    alt: true
+  },
+  {
+    exampleId: 'ex3',
+    linkSelector: '#ex3 span',
+    ariaLabel: true
+  }
+];
+
+ariaTest('link/link.html', 'link-role', async (t) => {
+
+  for (let i = 0; i < pageExamples.length; i++) {
+    let ex = pageExamples[i];
+    let linkLocator = t.context.By.css(ex.linkSelector);
+    let linkElement = await t.context.session.findElement(linkLocator);
+
+    t.is(
+      await linkElement.getAttribute('role'),
+      'link',
+      '[role="link"] attribute should exist on element select by: ' + ex.linkSelector
```

Comment:
Placing the assertion message here instead of in the test title means that a human-readable description of the test is only ever output if it fails. I'd prefer tests to always describe themselves.

Reply:
Also I want to be sure I understand what you are saying here -- the test title here is quick simple, it's `link/link.html [data-test-id="link-role"]`, constructed in ariaTest. The purpose that title is to tie  the description of the test to the description of the attribute/role/keypress that exists in the link/link.html example page itsself.

It sounds like you prefer the title of the test to be more verbose than this. What about titles like: `Test the 'role' attribute exists as documented in link/link.html [data-test-id="link-role"]`

Reply:
The first part of the title is great. I just meant that the actual description of what the test is testing only appears if the test fails. For example:
```bash
# success
  √ link » link/link.html [data-test-id="link-role"]
# failure
  × link » link/link.html [data-test-id="link-role"] [role="link"] attribute should exist on element select by: #ex1 span
```